### PR TITLE
ThreatsDataViews: Update isPrimary actions

### PR DESCRIPTION
**Pull request by dkmyta**
Description #
Moves access to actions toggles inline and hides the collapsible actions list

### Other information:

- [ ] Have you written new tests for your changes, if applicable?
- [x] Have you checked the E2E test CI results, and verified that your changes do not break them?
- [ ] Have you tested your changes on WordPress.com, if applicable (if so, you'll see a generated comment below with a script to run)?

## Jetpack product discussion

- https://github.com/Automattic/jetpack-scan-team/issues/1472

## Does this pull request change what data or activity we track or use?

- No

## Testing instructions:

* Checkout this branch
* Load storybook with `cd projects/js-packages/storybook && npm run storybook:dev`
* View the Threats Data View component
* Ensure the content of the actions section is positioned and styled according to designs in both Table and List views
* Review the code and ensure the implementation is correct

## Screenshots
Table View:
Screen Shot on 2024-10-22 at 13-57-16

List View:
Screen Shot 2024-10-22 at 14 32 29

### CODE REVIEW

File: projects/js-packages/components/components/threats-data-view/index.tsx [hunk 1]
```diff
@@ -359,7 +363,8 @@ export default function ThreatsDataView( {
 				id: 'ignore',
 				label: __( 'Ignore', 'jetpack' ),
 				isPrimary: true,
-				isDestructive: true,
+				icon: <span className={ styles.threat__action }>{ __( 'Ignore', 'jetpack' ) }</span>,
+				// isDestructive: true,
```
Comment (nateweller) [inline projects/js-packages/components/components/threats-data-view/index.tsx]:
```suggestion
```

File: projects/js-packages/components/components/threats-data-view/index.tsx [hunk 2]
```diff
@@ -379,7 +383,8 @@ export default function ThreatsDataView( {
 				id: 'un-ignore',
 				label: __( 'Unignore', 'jetpack' ),
 				isPrimary: true,
-				isDestructive: true,
+				icon: <span className={ styles.threat__action }>{ __( 'Un-ignore', 'jetpack' ) }</span>,
+				// isDestructive: true,
```
Comment (nateweller) [inline projects/js-packages/components/components/threats-data-view/index.tsx]:
```suggestion
```

File: projects/js-packages/components/components/threats-data-view/styles.module.scss [hunk 3]
```diff
@@ -16,6 +17,37 @@
...
+		}
+	}
+
+	.dataviews-view-table__actions-column .dataviews-item-actions .components-button.is-compact.has-icon {
+		width: fit-content;
+
+		&:focus,
+		&:focus-visible {
+			box-shadow: none;
+		}
```
Comment (nateweller) [inline projects/js-packages/components/components/threats-data-view/styles.module.scss:43]: Do we compensate for this with other focus styles?

Comment (dkmyta) [reply projects/js-packages/components/components/threats-data-view/styles.module.scss:43]: We do not - I wasn't quite sure to what extent we should compensate for all these things directly within the component or whether various parts should be handled in the implementation.

I removed the built in button element pseudo-class styles here and set the internal span element, text styles and a hover effect to match the colours in the designs, but there are other pieces in the component where we default to wp-admin style settings. Should we account for all of this here, or should we allow for the default and update everything where it's in use?

**Comment by dkmyta** [comment]
Closing in favour of adding the primary fixer actions to a dedicated column and keeping secondary action in the collapsible list as proposed in https://github.com/Automattic/jetpack/pull/39754.

**Pull request closed without merging**